delfin/drivers/pure/pure/pure_storage.py

```diff
@@ -0,0 +1,294 @@
+                    CONTROLLER_STATUS_MAP.get(status, constants.
+                                              ControllerStatus.UNKNOWN)
+                controllers_object['soft_version'] = controllers.get('version')
+                controllers_object['storage_id'] = self.storage_id
+                controllers_object['id'] = name
+                controllers_object['native_controller_id'] = name
+                controllers_object['location'] = ""
+                controllers_object['cpu_info'] = ""
+                controllers_object['memory_size'] = ""
+                list_controllers.append(controllers_object)
+        return list_controllers
+
+    def list_disks(self, context):
+        names = dict()
+        return_hardware = self.rest_handler.get_all_hardware()
+        if return_hardware:
+            for hardware in return_hardware:
+                hardware_name = dict()
+                hardware_name['speed'] = hardware.get('speed')
+                hardware_name['serial_number'] = hardware.get('serial')
+                hardware_name['model'] = hardware.get('model')
+                name = hardware.get('name')
+                names[name] = hardware_name
+
+        list_disk = []
+        return_disks = self.rest_handler.get_all_disk()
+        if return_disks:
+            for drive in return_disks:
+                disk = dict()
+                drive_name = drive.get('name')
+                disk['name'] = drive_name
+                physical_type = drive.get('type').lower()
+                if physical_type in constants.DiskPhysicalType.ALL:
+                    disk['physical_type'] = physical_type
+                else:
+                    disk['physical_type'] = constants.DiskPhysicalType.UNKNOWN
+                status = drive.get('status')
+                disk['status'] = StorageDriver.DISK_STATUS_MAP.\
+                    get(status, constants.DiskStatus.OFFLINE)
+                disk['storage_id'] = self.storage_id
+                disk['capacity'] = int(int(drive.get('capacity')) / units.Ki)
+                hardware_object = names.get(drive_name, {})
+
+                speed = hardware_object.get('speed')
+                if speed is None:
+                    disk['speed'] = ''
+                else:
+                    disk['speed'] = int(speed)
+                disk['model'] = hardware_object.get('model')
+                disk['serial_number'] = hardware_object.get('serial_number')
+
+                disk['native_disk_id'] = drive_name
+                disk['id'] = drive_name
+                disk['location'] = drive_name
+                disk['logical_type'] = ''
+                disk['native_disk_group_id'] = ""
+                disk['manufacturer'] = "pure"
+                disk['firmware'] = ""
+                disk['health_score'] = ""
+                list_disk.append(disk)
+        LOG.info("list_disk：%s" % (json.dumps(list_disk, ensure_ascii=False)))
```

Comment:
Why will we print this log info?

Reply:
The printed logs have been deleted. Procedure